src/sleep/sleep.c

```diff
@@ -40,6 +43,161 @@
 
 static char* arg_verb = NULL;
 
+static int read_fiemap(int fd, struct fiemap **ret) {
+        _cleanup_free_ struct fiemap *fiemap = NULL, *result_fiemap = NULL;
+        int extents_size;
+        struct stat statinfo;
+        uint32_t result_extents = 0;
+        uint64_t fiemap_start = 0, fiemap_length;
+        size_t fiemap_size = 0, result_fiemap_size = 0;
+
+        if (fstat(fd, &statinfo) != 0) {
+                log_debug_errno(errno, "Cannot determine file size (%s)\n %m",
+                                strerror(errno));
```

Comment:
%m already outputs the error string, no need to use strerror() anymore.

Please change this to:

```c
if (fstst(fd, &statinfo) < 0) 
        return log_debug_errno(errno, "Cannot determine file size: %m");
```

This takes benefit of the fact that log_debug_errno() returns the negative error you pass into it anyway